plugins/mac-capture/mac-sck-audio-capture.m

```diff
@@ -0,0 +1,386 @@
+					SCRunningApplication
+						*_Nonnull application,
+					NSUInteger idx, BOOL *_Nonnull stop) {
+					if ([application.bundleIdentifier
+						    isEqualToString:
+							    sc->
+							    application_id]) {
+						target_application =
+							sc->shareable_content
+								.applications
+									[idx];
+						*stop = TRUE;
+					}
+					return *stop;
+				}];
+		}
+		NSArray *target_application_array = [[NSArray alloc]
+			initWithObjects:target_application, nil];
+
+		content_filter = [[SCContentFilter alloc]
+			      initWithDisplay:target_display
+			includingApplications:target_application_array
+			     exceptingWindows:[[NSArray alloc] init]];
+
+		set_display_mode(sc, target_display);
+	} break;
+	}
+	os_sem_post(sc->shareable_content_available);
+	[sc->stream_properties setQueueDepth:8];
+
+	[sc->stream_properties setCapturesAudio:TRUE];
+	[sc->stream_properties setExcludesCurrentProcessAudio:TRUE];
+	[sc->stream_properties setChannelCount:2];
+
+	sc->disp = [[SCStream alloc] initWithFilter:content_filter
+				      configuration:sc->stream_properties
+					   delegate:nil];
+
+	//add a dummy video stream output to silence errors from SCK. frames are dropped by the delegate
+	NSError *error = nil;
+	BOOL did_add_output = [sc->disp addStreamOutput:sc->capture_delegate
+						   type:SCStreamOutputTypeScreen
+				     sampleHandlerQueue:nil
+						  error:&error];
+	if (!did_add_output) {
+		MACCAP_ERR(
+			"init_screen_stream: Failed to add stream output with error %s\n",
+			[[error localizedFailureReason]
+				cStringUsingEncoding:NSUTF8StringEncoding]);
+		[error release];
+		return !did_add_output;
+	}
+
+	did_add_output = [sc->disp addStreamOutput:sc->capture_delegate
+					      type:SCStreamOutputTypeAudio
+				sampleHandlerQueue:nil
+					     error:&error];
+	if (!did_add_output) {
+		MACCAP_ERR(
+			"init_audio_screen_stream: Failed to add audio stream output with error %s\n",
+			[[error localizedFailureReason]
+				cStringUsingEncoding:NSUTF8StringEncoding]);
+		[error release];
+		return !did_add_output;
+	}
+	os_event_init(&sc->disp_finished, OS_EVENT_TYPE_MANUAL);
+	os_event_init(&sc->stream_start_completed, OS_EVENT_TYPE_MANUAL);
+
+	__block BOOL did_stream_start = false;
+	[sc->disp startCaptureWithCompletionHandler:^(
+			  NSError *_Nullable error2) {
+		did_stream_start = (BOOL)(error2 == nil);
+		if (!did_stream_start) {
+			MACCAP_ERR(
+				"init_audio_screen_stream: Failed to start capture with error %s\n",
+				[[error localizedFailureReason]
+					cStringUsingEncoding:
+						NSUTF8StringEncoding]);
+			// Clean up disp so it isn't stopped
+			[sc->disp release];
+			sc->disp = NULL;
+		}
+		os_event_signal(sc->stream_start_completed);
+	}];
+	os_event_wait(sc->stream_start_completed);
+
+	MACCAP_ERR("init closing, returning %d\n", did_stream_start);
+	return did_stream_start;
+}
+
+static void sck_audio_capture_defaults(obs_data_t *settings)
+{
+	CGDirectDisplayID initial_display = 0;
```

Comment:
Please investigate if selecting a specific display makes a difference when it comes to audio output, i.e. if selecting a specific display, but another app outputs audio but is on another display, whether this influences the audio buffers we receive.

Reply:
I investigated this one during testing. Regardless of the display selected, SCK seems to pick up audio system-wide. I double checked with "Displays have separate Spaces" on and made sure that regardless of which display was selected, audio from other displays/Spaces would be picked up. I will double check and see if I missed any weird edge cases.

Reply:
Yeah, as far as I can tell, Display Capture will always pick up all system audio regardless of selected display. Even if the external monitor has an audio output and system audio is set to that external output, a Display Capture on the native display will still pick up that audio output, including for programs launched within and from that external Space (also vice versa). Tested with Music.app, Safari, and Firefox. I'll resolve this for now unless anyone can think of another edge case.